sherpa/models/parameter.py

```diff
@@ -19,6 +19,171 @@
+of the parameter (known as the "soft" limits). The default values
+for these are the 32-bit floating point maximum value, and it's
+negative:
+
+    >>> p.max
+    3.4028234663852886e+38
+    >>> p.min
+    -3.4028234663852886e+38
+
+Setting a value outside this range will raise a sherpa.utils.err.ParameterErr
+
+    >>> p.val = 1e40
+    ParameterErr: parameter model.eta has a maximum of 3.40282e+38
+
+These limits can be changed, as shown below, but they must lie
+within the `hard_min` to `hard_max` range of the parameter (the
+"hard" limits), which can not be changed:
+
+    >>> p.min = 0
+    >>> p.max = 10
+
+Freezing and thawing
+--------------------
+When fitting a model expression it is useful to be able to restrict
+the fit to a subset of parameters. This is done by only selecting
+those parameters which are not "frozen". This can be indicated by
+calling the `freeze` and `thaw` methods, or changing the `frozen`
+attribute directly:
+
+    >>> p.frozen
+    False
+    >>> p.freeze()
+    >>> p.frozen
+    True
+    >>> p.thaw()
+    >>> p.frozen
+    False
+
+Note that the `frozen` flag is used to indicate what parameters to
+vary in a fit, but it is still possible to directly change a parameter
+value when it is frozen:
+
+    >>> p.freeze()
+    >>> p.val = 6
+    >>> print(p)
+    val         = 6.0
+    min         = 0.0
+    max         = 10.0
+    units       =
+    frozen      = True
+    link        = None
+    default_val = 6.0
+    default_min = -3.4028234663852886e+38
+    default_max = 3.4028234663852886e+38
+
+Changing multiple settings at once
+----------------------------------
+The `set` method should be used when multiple settings need to be
+changed at once, as it allows for changes to both the value and
+limits, such as changing the value to be 20 and the limits to 8 to 30:
+
+    >>> p.val = 20
+    ParameterErr: parameter model.eta has a maximum of 10
+    >>> p.set(val=20, min=8, max=30)
+    >>> print(p)
+    val         = 20.0
+    min         = 8.0
+    max         = 30.0
+    units       =
+    frozen      = True
+    link        = None
+    default_val = 20.0
+    default_min = -3.4028234663852886e+38
+    default_max = 3.4028234663852886e+38
+
+Linking parameters
+------------------
+A parameter can be "linked" to another parameter, in which case the
+value of the parameter is calculated based on the link expression,
+such as being twice the other parameter:
+
+    >>> q = Parameter('other', 'beta', 4)
+    >>> p.val = 2 * p
+    >>> print(p)
+    val         = 8.0
+    min         = 8.0
+    max         = 30.0
+    units       =
+    frozen      = True
+    link        = (2 * other.beta)
+    default_val = 8.0
+    default_min = -3.4028234663852886e+38
+    default_max = 3.4028234663852886e+38
+
+The `link` attribute stores the expression:
+
+    >>> p.link
+    <BinaryOpParameter '(2 * other.beta)'>
+
+If the linked expression ends up to the parameters limits being
```

Comment:
```suggestion
If the linked expression ends violating the parameters limits 
```

Reply:
Ouch. That was a terrible sentence. I've put it out of its misery and re-wrote the paragraph (mainly just the first line). Hopefully it's more readable.